src/everything/prompts/args.ts

```diff
@@ -0,0 +1,41 @@
+import { z } from "zod";
+import { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
+
+/**
+ * Register a prompt with arguments
+ * - Two arguments, one required and one optional
+ * - Combines argument values in the returned prompt
+ *
+ * @param server
+ */
+export const registerArgumentsPrompt = (server: McpServer) => {
+  // Prompt arguments
+  const promptArgsSchema = {
+    city: z.string().describe("Name of the city"),
+    state: z.string().describe("Name of the state").optional(),
+  };
+
+  // Register the prompt
+  server.registerPrompt(
+    "args-prompt",
```

Comment:
nit: The convention I've generally seen for prompts and tool names is `snake_case` - is there a reason we're using kebab case? Seems good to follow conventions where we can in case agents use this to see how to implement particular features etc. and pick up on patterns.

Reply:
The `snake_case` convention mostly comes from Python server development, because the convention matches Python idioms. The point of deviating here is that it illustrates that snake is a convention, but tool names can be whatever you want them to be within the allowed parameters. A `kebab-case` primitive can line up with the file name that registers making for a single naming strategy. Same benefit of having one case pattern throughout the project, but with TS friendly idioms. It's a design choice one can make.